qutebrowser/completion/models/filepathcategory.py

```diff
@@ -0,0 +1,63 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Completion category for filesystem paths."""
+
+import glob
+import os
+from typing import Any
+import urllib.parse
+
+from PyQt5.QtCore import (QAbstractListModel, QModelIndex)
+
+
+class FilePathCategory(QAbstractListModel):
+    """Represent filesystem paths matching a pattern."""
+
+    def __init__(self, name: str):
+        super().__init__()
```

Comment:
```suggestion
    def __init__(self, name: str, parent: QObject = None) -> None:
        super().__init__(parent)
```

to support Qt's object trees properly (to e.g. ensure this is cleaned up when its parent is)